File: Algorithms/35. Search Insert Position.rs
impl Solution {
    pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
        let mut i = 0;
        if target < nums[0] { return 0; }
        while i < nums.len() {
            if nums[i] == target { return i as i32;}
            if i < nums.len() -1 && nums[i + 1] > target { return (i + 1)  as i32;}
            i += 1;
        }
        i  as i32
    }
}

File: Algorithms/35. Search Insert Position.ts
function searchInsert(nums: number[], target: number): number {
  let left: number = 0;
  let right: number = nums.length;
  while (left <= right) {
    const mid: number = Math.floor((left + right) / 2);
    if (nums[mid] === target) {
      return mid;
    } else if (nums[mid] < target) {
      left = mid + 1;
    } else {
      right = mid - 1;
    }
  }

  return left;
}
